test/testdata/compiler/intrinsics/identities.rb

```diff
@@ -56,3 +62,9 @@ def self.symbol_to_sym(x)
 p M.integer_to_int(20)
 
 p M.symbol_to_sym(:foo)
+
+class StringSubclass < String
+end
```

Comment:
Shouldn't this class be defining its own `to_s` method to ensure that our intrinsic implementation isn't just calling `rb_str_to_s` directly?

Reply:
We should test both. The default implementation of `String#to_s` will do something different for subclasses of `String`, which is what this case is testing.